dynamic_models/models.py

```diff
@@ -59,7 +70,10 @@ def get_model_name(cls, name):
 
     @property
     def db_table(self):
-        parts = (self.app_label, slugify(self.name).replace("-", "_"))
+        if self.db_table_name is None:
+            parts = (self.app_label, slugify(self.name).replace("-", "_"))
+        else:
+            parts = (slugify(self.db_table_name).replace("-", "_"),)
         return "_".join(parts)
```

Comment:
**suggestion**: if there is an overridden table name then there's no need to join the parts. We could refactor this a little to be very explicit that we are overriding a default value here

```suggestion
        return self.db_table_name if self.db_table_name else self._default_db_table_name() 
    
    def _default_db_table_name(self):
        safe_name = slugify(self.name).replace("-", "_")
        return f"{self.app_label}_{safe_name}"
```

I was curious looking at the Django source for default `db_table` and unfortunately they don't have it available as a utility method we can use, but it could give you an idea for how they handle platform specific length constraints

https://github.com/django/django/blob/main/django/db/models/options.py#L228-L232

Either way is fine with me. I think the current way should keep working most of the time, but will raise errors if a user has a table name with a value greater than the platform's max length. The way Django does it is probably most correct but definitely more effort

Reply:
i would keep this as is now, we can open an improvement 👍 
anyway yes, it may be the safest way to handle it